nselib/smb2.lua

```diff
@@ -0,0 +1,410 @@
+---
+-- Implements the Server Message Block (SMB) protocol version 2 and 3.
+-- 
+-- The implementation extends smb.lua to support SMB dialects 2.02, 2.10, 3.0,
+--  3.02 and 3.11. This is a work in progress and not all commands are
+--  implemented yet. Features/functionality will be added as the scripts
+--  get updated. I tried to be consistent with the current implementation of
+--  smb.lua but some fields may have changed name or don't exist anymore.
+--
+-- TODO:
```

Comment:
Is this TODO list current? It probably doesn't belong in the NSEdoc. Please make a separate Github Issue for each item that you think really ought to be done.